lib/posix/options/pthread.c

```diff
@@ -869,6 +869,46 @@ int pthread_setschedparam(pthread_t pthread, int policy, const struct sched_para
 	return ret;
 }
 
+/**
+ * @brief Set thread scheduling priority.
+ *
+ * See IEEE 1003.1
+ */
+int pthread_setschedprio(pthread_t thread, int prio)
+{
+	int ret = 0;
+	int new_prio = K_LOWEST_APPLICATION_THREAD_PRIO;
+	struct posix_thread *t = NULL;
+	int policy = 0;
```

Comment:
I think only `new_prio` needs to be initialized?

Reply:
```suggestion
	int ret = 0;
	int new_prio = K_LOWEST_APPLICATION_THREAD_PRIO;
	struct posix_thread *t = NULL;
	int policy;
```
ret and t should be initialized I think.

Reply:
`ret` isn't used before L885 and will be assigned at L885

the initialization of `t` is probably required here, since it is used in L906

Reply:
```suggestion
	int ret;
	int new_prio = K_LOWEST_APPLICATION_THREAD_PRIO;
	struct posix_thread *t = NULL;
	int policy;
```